docs/guides/admin/docs/configuration/lifecycle-management.md

```diff
@@ -0,0 +1,54 @@
+LifeCycle Management
+====================
+
+The goal of LifeCycle Management is to allows users to automate administrative tasks that are specific to their use
+case. For example:
+
+" We want to automatically retract videos of lecture recordings that have been uploaded during the winter semester 2024
+at the end of the semester. Lecture recordings are always organized in a series. Conferences should be excluded from
+this, as well as all lectures by the VIP Albert Einstein."
+
+To that end, user can create LifeCycle Policies. They allow users to define
+
+- what should happen (action),
+- to what it should happen to (target/object) and
+- when it should happen (time).
+
+The different components of a policy are described below in greater detail.
+
+## Policy Action
+
+For now there is only one action that can be specified, called "START_WORKFLOW". “START_WORKFLOW” takes a workflow
+definition ID (and workflow configuration) to define which workflow should be run. For example, in order to retract
+events we would want to run the “retract” workflow. What exactly the “retract” workflow does is then up to the admins
+of each institution, i.e. can be customized by each institution as they see fit.
+
+## Policy Target Type
+
+Determines which entity the policy should be applied to. The two entities we have in Opencast are events
+or series. Currently only EVENTS are supported.
+
+## Policy Filters
+
+Criteria that further narrow down which entities the policy should be applied to. Filters make use of an entities'
+metadata. A policy can have none, one or multiple filters. For example, if the policy should only run on a certain
+series it may have the filter “Series identifier is equal to ID-av-portal”. To have the policy apply to all videos of a
+semester it may have the two filters “Start Date > 2023-12-20” and “Start Date < 2024-02-30” to delete all videos
+from December 20 2023 to February 30 2024.
+
+
+## Policy timing
+
+The point in time when an action should be performed. There are three types of action dates:
+
+- SPECIFIC_DATE: A fixed date, for example 2024-02-31 at 11:30. This means the policy will be applied exactly once at
+  that point in time. Useful for example for retracting videos at the end of a semester.
+- REPEATING: A periodic date, for example “Every day at 02:00 AM”. This means the policy will be applied to all events
```

Comment:
Editing is currently allowed, but not always in your best interest. When reaching the specified action date, a policy will create "lifecycle tasks" for each entity (currently only events) matching the filters. The tasks remember the id of the policy that created them, and a policy will not create a new task for an entity that already has a task with the policies id associated with it. Or in short, the same policy won't run on the same entity twice.

Have I not made mention of this in the docs? Seems important to know...

Reply:
I didn't catch it if you did.

I don't know what your client is looking for, but I would almost be tempted to *not* allow editing.  The fact that it might not be in your best interest is a code smell IMO.  Not allowing editing simplifies some edge cases at the very least.  Allowing the user to edit a policy with a action date in the past, and reschedule it to be in the future leads to confusion for the user - why hasn't this been reapplied to matching events?  And now you have to keep track of which events it's already been applied to *and* expose that to the user in the UI.

Reply:
Regarding the create+delete cycle, that's what UI magic is for :)  I wouldn't expect the user to do this, but the *backend* could require it to simplify the internal logic.  Or maybe that just moves the problem to the front end!

A reset would likely also solve the issue, at least on first glance.  You've worked on this more than I have, and I'm not picky - just trying to prevent issues in the future when people really start to use this.